Neos.Flow/Classes/Reflection/ReflectionService.php

```diff
@@ -1800,11 +1797,20 @@ protected function convertParameterReflectionToArray(ParameterReflection $parame
 
         $parameterType = $parameter->getType();
         if ($parameterType !== null) {
+            // TODO: This needs to handle ReflectionUnionType
             $parameterType = ($parameterType instanceof \ReflectionNamedType) ? $parameterType->getName() : $parameterType->__toString();
         }
-        if ($parameter->getClass() !== null) {
+        if ($parameterType !== null && !TypeHandling::isSimpleType($parameterType)) {
             // We use parameter type here to make class_alias usage work and return the hinted class name instead of the alias
             $parameterInformation[self::DATA_PARAMETER_CLASS] = $parameterType;
+        } elseif ($parameterType === 'array') {
+            $parameterInformation[self::DATA_PARAMETER_ARRAY] = true;
```

Comment:
```suggestion
            $parameterInformation[self::DATA_PARAMETER_ARRAY] = true;
            $parameterInformation[self::DATA_PARAMETER_TYPE] = $parameterType;
```
Maybe this is missing to stay fully b/c - was previously set below with "$parameter->isArray() ? 'array' : 'mixed'"

Reply:
Yes, please.

Reply:
Okay, I get why this breaks now: We set $parameterType here, where it is just 'array'. But we need to actually use the expanded parameter type information from the annotation below to keep the annotated element type:
`$parameterType = $this->expandType($method->getDeclaringClass(), $explodedParameters[0]);`

The DATA_PARAMETER_TYPE will then be set with the `if (!isset($parameterInformation[self::DATA_PARAMETER_TYPE]) && $parameterType !== null) {` condition